### ISSUE TITLE
Specific price for product: Error with comma/period when setting percentage discount with decimals

### ISSUE THREAD

**Issue by LarsGMortensen** [PrestaShop/PrestaShop#33876]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

In PrestaShop 8.1.1 with Danish language (and possibly others?) where we use a comma rather than a period as a decimal separator.
when setting a specific price; percentage discount on a product with decimals (i.e. 19,19% discount)
PrestaShop will not validate the input on the first attempt, but makes it right when trying again.

I seems to be related to comma vs. period as the decimal separator.

Please see this screen-recording for visual explanation (its an animated GIF so it will loop automatically):

![3hfE7bOU5j](https://github.com/PrestaShop/PrestaShop/assets/26773745/5cd6aadd-3dfb-4641-95bd-0d1d4dc314fd)

### Steps to reproduce

1. In PrestaShop 8.1.1 with Danish language (and possibly other languages where commas are used as decimal separator)
2. Edit a product (I have reproduced the error with standard products and products with combinations)
3. Go the "Price" tab
4. Set a specific price with percentage of a number with decimals (i.e. 19,19% discount) and click save.
5. Notice the validation-error as PrestaShop would not accept your input of 19,19%

Please see my screen-recording above for visual explanation.

### PrestaShop version(s) where the bug happened

8.1.1

### PHP version(s) where the bug happened

7.4.33

**Comment by Hlavtox** [comment]
Yeah, in Czech, we also use decimal comma. It's extremely annoying that Prestashop doesn't handle it.

**Comment by hibatallahAouadni** [comment]
Hello @LarsGMortensen 

Your issue seems similar to this one #24889 could you please check it and feedback.
Waiting for your feedback.

Thanks!

**Comment by LarsGMortensen** [comment]
Hi @hibatallahAouadni 
I dont think its the same. Also product pages look differently and the other ticket is closed, so that old issue must have been fixed, so this must be a new bug?
But anyway; I am pretty sure this is a smaller mistake somewhere in the handling hos . vs. , as decimal delimiter?

**Comment by sallemiines** [comment]
Hello @LarsGMortensen 

I didn't manage to reproduce the issue with 8.1.x, see the attached screen record below:

[issue_33876.webm](https://github.com/PrestaShop/PrestaShop/assets/31663369/d3d05439-253e-4825-91e5-cdec6d9334d6)

Please check and feedback.

Thanks!

**Comment by LarsGMortensen** [comment]
Dear @sallemiines 
Could you please see the screen-recording that I posted in the initial comment above :)

I am NOT talking about "price" (the base price of the product), but "specific price" (discounts on the product).
I am sure it will all make better sense if you would actually spend a moment watching the screen-recording that I posted with the initial comment above :) Thanks.

**Comment by MhiriFaten** [comment]
Hello,

I reproduce the issue with PrestaShop version 8.1.x and develop.

[Untitled_ Sep 14, 2023 3_56 PM.webm](https://github.com/PrestaShop/PrestaShop/assets/111756615/b8b81fc9-81b5-4a72-aed3-b496b999a85d)


Thank you

**Comment by bynante** [comment]
This still affects merchants in Q4 2025 on the latest 8.2.x branch when using locales with comma decimal separators. Is there any update on this bug or an ETA for a fix?

**Comment by jolelievre** [comment]
Maybe you can try and add `PercentType` into the handled types by `CommaTransformerExtension` I'm not sure but it's worth trying

**Comment by boo-code** [comment]
I could not pin this one down, but I can rule out the three explanations it most looks like, which should save whoever picks it up the same dead ends.

**The comma normaliser handles the reported input correctly.** Running the real `transform()` from `js/app/utils/number-comma-transformer.js` in node:

```
19,19      -> "19.19"
1.234,56   -> "1234.56"
1,234.56   -> "1234.56"
-19,19     -> "-19.19"
0,5        -> "0.5"
```

**The field is covered by it.** The reduction value is a `MoneyType`, and `CommaTransformerExtension::getExtendedTypes()` returns both `NumberType` and `MoneyType`, so the input carries `js-comma-transformer` and the listener registered in `theme.ts` applies on `change`.

**It is not the browser blanking an invalid number.** `number_widget` and the money widget both render `type="text"` in this theme, so a comma is not rejected by the input itself and `value` is not emptied.

**And the server accepts either form.** `NumberToLocalizedStringTransformer::reverseTransform()` under `en`, `da` and `fr`:

```
Danish   locale, input 19,19 -> 19.19
Danish   locale, input 19.19 -> 19.19
```

So the value is normalised correctly on the way out and parsed correctly on the way in. What is left is *when* the normalisation runs relative to whatever rejects the input: the listener fires on `change`, so anything that reads the field earlier — an `input`/`keyup` handler, or a submit path that serialises from a JS model rather than the DOM — sees the raw `19,19`. That is a timing question I cannot settle by reading, and I have no back office session here to watch it in.

@LarsGMortensen, two things would narrow it immediately if you still have that shop: the exact text of the error shown on the first attempt, and whether it also happens when you paste `19,19` rather than typing it (pasting fires `input` but not necessarily `change` before you click). It would also help to know whether it is the product page's specific prices or the standalone catalog price rules form, since those are separate screens.

Nothing prepared for this one; I would rather not guess at a fix while the trigger is unconfirmed.
